src/MongoDB.Bson/ObjectModel/BsonRegularExpression.cs

```diff
@@ -286,5 +286,13 @@ public override string ToString()
             var escaped = (_pattern == "") ? "(?:)" :_pattern.Replace("/", @"\/");
             return string.Format("/{0}/{1}", escaped, _options);
         }
+
+        // private methods
+        private string SortOptions(string options)
+        {
+            var sorted = options.ToCharArray();
+            Array.Sort(sorted);
+            return new string(sorted);
+        }
```

Comment:
Consider refactoring to avoid unnecessary memory allocations:

    private string SortOptionsIfNecessary(string options)
    {
        if (IsAlreadySorted(options))
        {
            return options;
        }
        else
        {
            var sorted = options.ToCharArray();
            Array.Sort(sorted);
            return new string(sorted);
        }

        bool IsAlreadySorted(string value)
        {
            for (var i = 0; i < value.Length - 1; i ++)
            {
                if (value[i] > value[i + 1])
                {
                    return false;
                }
            }

            return true;
        }
    }

Reply:
Done.